### ISSUE TITLE
Conda update conda - unexpected error (Win7)

### ISSUE THREAD

**Issue by LilithElina** [conda/conda#3264]
Hi,
I am trying to update conda on my Windows 7 laptop at work. Here is what happens:

> C:\Anaconda>conda update conda
> Fetching package metadata: C:\Anaconda\lib\site-packages\requests\packages\urlli
> b3\util\ssl_.py:315: SNIMissingWarning: An HTTPS request has been made, but the
> SNI (Subject Name Indication) extension to TLS is not available on this platform
> . This may cause the server to present an incorrect TLS certificate, which can c
> ause validation failures. For more information, see https://urllib3.readthedocs.
> org/en/latest/security.html#snimissingwarning.
>   SNIMissingWarning
> C:\Anaconda\lib\site-packages\requests\packages\urllib3\util\ssl_.py:120: Insecu
> rePlatformWarning: A true SSLContext object is not available. This prevents urll
> ib3 from configuring SSL appropriately and may cause certain SSL connections to
> fail. For more information, see https://urllib3.readthedocs.org/en/latest/securi
> ty.html#insecureplatformwarning.
>   InsecurePlatformWarning
> C:\Anaconda\lib\site-packages\requests\packages\urllib3\util\ssl_.py:120: Insecu
> rePlatformWarning: A true SSLContext object is not available. This prevents urll
> ib3 from configuring SSL appropriately and may cause certain SSL connections to
> fail. For more information, see https://urllib3.readthedocs.org/en/latest/securi
> ty.html#insecureplatformwarning.
>   InsecurePlatformWarning
> C:\Anaconda\lib\site-packages\requests\packages\urllib3\util\ssl_.py:120: Insecu
> rePlatformWarning: A true SSLContext object is not available. This prevents urll
> ib3 from configuring SSL appropriately and may cause certain SSL connections to
> fail. For more information, see https://urllib3.readthedocs.org/en/latest/securi
> ty.html#insecureplatformwarning.
>   InsecurePlatformWarning
> C:\Anaconda\lib\site-packages\requests\packages\urllib3\util\ssl_.py:120: Insecu
> rePlatformWarning: A true SSLContext object is not available. This prevents urll
> ib3 from configuring SSL appropriately and may cause certain SSL connections to
> fail. For more information, see https://urllib3.readthedocs.org/en/latest/securi
> ty.html#insecureplatformwarning.
>   InsecurePlatformWarning
> ....
> .Solving package specifications: ..
> An unexpected error has occurred, please consider sending the
> following traceback to the conda GitHub issue tracker at:

I also get the following traceback:

> Traceback (most recent call last):
>   File "C:\Anaconda\Scripts\conda-script.py", line 5, in <module>
>     sys.exit(main())
>   File "C:\Anaconda\lib\site-packages\conda\cli\main.py", line 139, in main
>     args_func(args, p)
>   File "C:\Anaconda\lib\site-packages\conda\cli\main.py", line 146, in args_func
> 
>     args.func(args, p)
>   File "C:\Anaconda\lib\site-packages\conda\cli\main_update.py", line 64, in exe
> cute
>     install.install(args, parser, 'update')
>   File "C:\Anaconda\lib\site-packages\conda\cli\install.py", line 334, in instal
> l
>     update_deps=args.update_deps)
>   File "C:\Anaconda\lib\site-packages\conda\plan.py", line 413, in install_actio
> ns
>     pkgs = r.install(specs, [d + '.tar.bz2' for d in linked], update_deps=update
> _deps)
>   File "C:\Anaconda\lib\site-packages\conda\resolve.py", line 875, in install
>     pkgs = self.solve(specs, len0=len0, returnall=returnall)
>   File "C:\Anaconda\lib\site-packages\conda\resolve.py", line 918, in solve
>     solution = C.sat(constraints, True)
>   File "C:\Anaconda\lib\site-packages\conda\logic.py", line 422, in sat
>     solution = pycosat.solve(clauses, vars=self.m, prop_limit=limit)
> TypeError: list expected

Here's my conda info:

> C:\Anaconda>conda info
> Current conda install:
> 
>              platform : win-64
>         conda version : 4.0.5
>   conda-build version : 1.3.1
>        python version : 2.7.6.final.0
>      requests version : 2.9.1
>      root environment : C:\Anaconda  (writable)
>   default environment : C:\Anaconda
>      envs directories : C:\Anaconda\envs
>         package cache : C:\Anaconda\pkgs
>          channel URLs : https://repo.continuum.io/pkgs/free/win-64/
>                         https://repo.continuum.io/pkgs/free/noarch/
>                         https://repo.continuum.io/pkgs/pro/win-64/
>                         https://repo.continuum.io/pkgs/pro/noarch/
>           config file : C:\Users\spo12\.condarc
>     is foreign system : False

What can I do to avoid that?

**Comment by HugoTian** [comment]
Thanks for your info, an easy fix will be remove the anaconda folder , and use the new miniconda installer to install latest conda (version 4.1.11)

Your version is 4.0.5 and I am not quite familiar with that version, and it may take some time to get the solution.

**Comment by LilithElina** [comment]
The command you posted gives the same error as the other things I've tried, so I guess I'm up for a new install.

Just one more question: what I actually wanted to do was get Junyper to work with an R kernel, but apparently `junyper`, as a command, is not working for me. I guessed my conda/Anaconda install was too old, because when I start `ipython notebook` it does say Junyper right on top of the page.
Now that updating doesn't work, would it make sense to simply make a clean, new install of the whole Anaconda package?
